packages/backend/src/handlers/triggers/updateDiscordRole.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+import Discord, { Role } from 'discord.js';
+
+import { CONFIG } from '../../config';
+import { AccountType_Enum, Player, PlayerRank_Enum } from '../../lib/autogen/hasura-sdk';
+import { client } from '../../lib/hasuraClient';
+import { TriggerPayload } from './types';
+
+export interface UpdateRole {
+  playerId: string;
+  previousRole: string | undefined;
+  newRole: string;
+}
+
+export const updateDiscordRole = async (
+  payload: TriggerPayload<Player>,
+) => {
+  const {old: oldPlayer, new: newPlayer} = payload.event.data;
+
+  try {
+    if (newPlayer == null) return;
+   
+    const getPlayerResponse = await client.GetPlayer({playerId: newPlayer.id});
+    const discordPlayerAccount = getPlayerResponse.player_by_pk?.Accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordPlayerAccount?.identifier == null) return;
+
+    const newRank = newPlayer?.rank;
+
+    if (newRank == null) return;
+   
+    // look up guild by guildname = 'metagame' (for now), 
+    const getGuildResponse = await client.GetGuild({guildname: 'metafam'});
+    const discordAccount = getGuildResponse.guild[0]?.guild_accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordAccount == null) return;
+
+    // call discord API. We'll need serverId, playerId, and roleIds
+    const discordClient = new Discord.Client();
+    discordClient.login(CONFIG.discordBotToken);
+
+    // todo add jsonb field to guild and populate ranks, e.g. { ranks: [] }
```

Comment:
I want to make "discord roles as a source of truth for guild membership" and display all the roles a player has in each guild on their profiles, so the idea of "ranks" can just be subsumed by the "roles" (with the difference being we have specific logic to update ranks automatically based on XP and update roles in discord, whereas the other roles would just be synced from discord into our DB).

However I know that trying to keep two sources of truth in sync (DB and what role is set in discord) is never a good time, so was thinking we could even fetch the roles directly from discord API and remote join them to the Player table. Happy to do syncing for now but I can imagine it causing issues in the future

Reply:
Yes, I was going to tackle that in #184. At that point this will probably get refactored.